### DISCUSSION TITLE
`jnp.stack()` OOM on TPUv3-8

### DISCUSSION THREAD

**Discussion by AakashKumarNain** [jax-ml/jax#20167] [Q&A]
I am using [Equinox](https://github.com/patrick-kidger/equinox) for one of my workflows. All Equniox modules are just Pytrees. I have a list of these `Pytree` modules like this:

```python
layers = [some_module for key in keys]  # This works
```

The total parameters count in `layers` is `6979584000` where each param is float16 dtype. When I try to stack this into a one large array, I get OOM error on TPUv3-8.

```python
layers = [some_module for key in keys] 
layers = jax.tree_util.tree_map(lambda *x: jnp.stack(x), *layers) # Throws OOM
```

Can someone provide any pointers to why is this happening?

**Comment by jakevdp** [comment]
I'm not sure what you mean by "total parameters count": to clarify, how many arrays do you have, and how many elements are in each?

**Reply by AakashKumarNain** [reply]
I apologize for the vagueness. I will rephrase it in a better way with an example
